support_bundle/get_support_bundle.sh

```diff
@@ -11,216 +11,367 @@ catch() {
+        )
+        curl "${PARAMS[@]}" >${LOG_DIR}/metrics/agent_version_metric_limits.json || echo "Curl failed collecting agent_version_metric_limits.json data!" && true
+}
+
+get_metrics() {
+# function used to get metric JSON data for particular metrics we are interested in from the agent
+# arguments:
+# 1 - metric_name
+# 2 - segment_by
+metric="${1}"
+segment_by="${2}"
+
+        PARAMS=(
+          -sk --location --request POST "${API_URL}/api/data/batch?metricCompatibilityValidation=true&emptyValuesAsNull=true"
+          --header 'X-Sysdig-Product: SDC'
+          --header "Authorization: Bearer ${API_KEY}"
+          --header 'Content-Type: application/json'
+          -d "{\"requests\":[{\"format\":{\"type\":\"data\"},\"time\":{\"from\":${FROM_EPOCH_TIME}000000,\"to\":${TO_EPOCH_TIME}000000,\"sampling\":600000000},\"metrics\":{\"v0\":\"${metric}\",\"k0\":\"timestamp\",\"k1\":\"${segment_by}\"},\"group\":{\"aggregations\":{\"v0\":\"avg\"},\"groupAggregations\":{\"v0\":\"avg\"},\"by\":[{\"metric\":\"k0\",\"value\":600000000},{\"metric\":\"k1\"}],\"configuration\":{\"groups\":[{\"groupBy\":[]}]}},\"paging\":{\"from\":0,\"to\":9},\"sort\":[{\"v0\":\"desc\"}],\"scope\":null,\"compareTo\":null}]}'"
+        )
+        curl "${PARAMS[@]}" >${LOG_DIR}/metrics/${metric}_${segment_by}.json || echo "Curl failed collecting ${metric}_${segment_by} data!" && true
+}
+
+main() {
+    # Check for supplied namespace, kube context, and flags
+    if [[ -z ${NAMESPACE} ]]; then
+        NAMESPACE="sysdig"
+    fi
+
+    if [[ ! -z ${CONTEXT} ]]; then
+        CONTEXT_OPTS="--context=${CONTEXT}"
+    fi
+
+    if [[ ! -z ${SINCE} ]]; then
+        SINCE_OPTS="--since ${SINCE}"
+    fi
+
+    # Set options for kubectl commands
+    KUBE_OPTS="--namespace ${NAMESPACE} ${CONTEXT_OPTS}"
+
+    #verify that the provided namespace exists
+    KUBE_OUTPUT=$(kubectl ${KUBE_OPTS} get namespace ${NAMESPACE}) || true
+
+    # Check that the supplied namespace exists, and if not, output current namespaces
+    if [[ "$(echo "$KUBE_OUTPUT" | grep -o "^sysdig " || true)" != "${NAMESPACE} " ]]; then
+        echo "We could not determine the namespace. Please check the spelling and try again";
+        echo "kubectl ${KUBE_OPTS} get ns";
+        echo "$(kubectl ${KUBE_OPTS} get ns)";
+    fi
+    # If API key is supplied, collect streamSnap, Index settings, and fastPath settings
+    if [[ ! -z ${API_KEY} ]]; then
+        API_URL=$(kubectl ${KUBE_OPTS} get cm sysdigcloud-config -o yaml | grep -i api.url: | head -1 | awk '{print$2}')
+        curl -ks -H "Authorization: Bearer ${API_KEY}" -H "Content-Type: application/json" "${API_URL}/api/license" >> ${LOG_DIR}/license.json
+        curl -ks -H "Authorization: Bearer ${API_KEY}" -H "Content-Type: application/json" "${API_URL}/api/admin/customer/1/streamsnapSettings" >> ${LOG_DIR}/streamSnap_settings.json
```

Comment:
this is fine for now, but note that we have had customers in the past use a different customer id than `1`